### PR TITLE
Added aria attributes to Menu component to support screen readers.

### PR DESCRIPTION
**Pull request by TiagoJustino**
This PR improves the screen reader support for the Menu component.
- the 'menu' and 'menuitem' roles were added.
- aria-expanded, aria-selected and aria-activedescendant attributes added

### CODE REVIEW

File: src/js/components/Menu.js [hunk 1]
```diff
@@ -39,6 +39,18 @@ var MenuDrop = React.createClass({
       down: this._onDownKeyPress
     };
     this.startListeningToKeyboard(this._keyboardHandlers);
+    var menuItems = this.refs.navContainer.getDOMNode().childNodes;
+    for (var i = 0; i < menuItems.length; i++) {
+      menuItems[i].setAttribute('role', 'menuitem');
+      
+      menuItems[i].setAttribute('id', menuItems[i].getAttribute('data-reactid'));
```
Comment (ericsoderberghp) [inline src/js/components/Menu.js]: What if the caller has already set an id attribute on the menu item? I wouldn't want Grommet to overwrite it.

Comment (TiagoJustino) [inline src/js/components/Menu.js]: You're right. I missed this check.

File: src/js/components/Menu.js [hunk 2]
```diff
@@ -39,6 +39,18 @@ var MenuDrop = React.createClass({
       down: this._onDownKeyPress
     };
     this.startListeningToKeyboard(this._keyboardHandlers);
+    var menuItems = this.refs.navContainer.getDOMNode().childNodes;
+    for (var i = 0; i < menuItems.length; i++) {
+      menuItems[i].setAttribute('role', 'menuitem');
```
Comment (ericsoderberghp) [inline src/js/components/Menu.js]: Is this role appropriate for nested Menu items? For things like headers?
I wonder if we should be more selective in what we place this role attribute on.

Comment (TiagoJustino) [inline src/js/components/Menu.js]: That makes sense! How do you think it'd be the best way to filter this? by tag? or by some other property?

Comment (ericsoderberghp) [inline src/js/components/Menu.js]: Perhaps filtering to just <a>, .anchor, or .button children?

File: src/js/components/CheckBox.js [hunk 3]
```diff
@@ -13,20 +14,25 @@ var CheckBox = React.createClass({
...
     }
     if (this.props.className) {
       classes.push(this.props.className);
     }
+    
     return (
-      <label className={classes.join(' ')}>
-        <input className={CLASS_ROOT + "__input"}
+      <label className={classes.join(' ')}
+        aria-describedby={this.props.ariaDescribedby}
```
Comment (alansouzati) [inline src/js/components/CheckBox.js]: Can you add `ariaDescribedBy` in CheckboxDoc ?

File: src/js/components/Menu.js [hunk 4]
```diff
@@ -39,6 +39,25 @@ var MenuDrop = React.createClass({
...
+      
+      if (!menuItems[i].getAttribute('id')) {
+        menuItems[i].setAttribute('id', menuItems[i].getAttribute('data-reactid'));
+      }
+      // aria-selected informs AT which menu item is selected for that menu container.
+      if (classes.indexOf('active') > -1) {
+        menuItems[i].setAttribute('aria-selected', true);
+      } else {
+        menuItems[i].setAttribute('aria-selected', false);
```
Comment (alansouzati) [inline src/js/components/Menu.js]: can you add this as: `menuItems[i].setAttribute('aria-selected', classes.indexOf('active') > -1);`

File: src/js/components/Menu.js [hunk 5]
```diff
@@ -46,27 +65,52 @@ var MenuDrop = React.createClass({
...
+    var menuItems = this.refs.navContainer.getDOMNode().childNodes;
     if (!this.activeMenuItem) {
-      var menuItems = this.refs.navContainer.getDOMNode().childNodes;
       var lastMenuItem = menuItems[menuItems.length - 1];
       this.activeMenuItem = lastMenuItem;
     } else if (this.activeMenuItem.previousSibling) {
       this.activeMenuItem = this.activeMenuItem.previousSibling;
     }
 
+    var classes = this.activeMenuItem.className.split(/\s+/);
```
Comment (alansouzati) [inline src/js/components/Menu.js]: //want to skip items of the menu that are not focusable.

File: src/js/components/Menu.js [hunk 6]
```diff
@@ -46,27 +65,52 @@ var MenuDrop = React.createClass({
...
+    var tagName = this.activeMenuItem.tagName.toLowerCase();
+    if (tagName !== 'button' && tagName !== 'a' && classes.indexOf('check-box') === -1) {
+      if (this.activeMenuItem === menuItems[0]) {
+        return true;
+      } else {
+        return this._onUpKeyPress(event);
```
Comment (alansouzati) [inline src/js/components/Menu.js]: //move up to the next item (possible next focusable item).

File: src/js/components/Menu.js [hunk 7]
```diff
@@ -46,27 +65,52 @@ var MenuDrop = React.createClass({
...
+    event.preventDefault();
+    var menuItems = this.refs.navContainer.getDOMNode().childNodes;
     if (!this.activeMenuItem) {
-      this.activeMenuItem = this.refs.navContainer.getDOMNode().childNodes[0];
+      this.activeMenuItem = menuItems[0];
     } else if (this.activeMenuItem.nextSibling) {
       this.activeMenuItem = this.activeMenuItem.nextSibling;
     }
 
+    var classes = this.activeMenuItem.className.split(/\s+/);
```
Comment (alansouzati) [inline src/js/components/Menu.js]: same documentation as up key

File: src/js/components/Menu.js [hunk 8]
```diff
@@ -211,6 +255,23 @@ var Menu = React.createClass({
...
       });
+
+      controlElement.setAttribute('role', 'menu');
+      var expanded = this.state.state === 'expanded';
+      controlElement.setAttribute('aria-expanded', expanded);
+      if (this.props.label) {
+        controlElement.setAttribute('aria-label', this.props.label);
+      } else if (this.props.icon) {
+        try {
+          var icon = controlElement.querySelectorAll('.control-icon')[0];
```
Comment (alansouzati) [inline src/js/components/Menu.js]: can you replace it by `controlElement.getElementsByClassName` ?

File: src/js/components/index/IndexFilters.js [hunk 9]
```diff
@@ -128,19 +128,18 @@ var IndexFilters = React.createClass({
...
+          <CheckBox key={attribute.attribute + '-all'}
+            className={CLASS_ROOT + "__filter-value"}
             id={attribute.attribute + '-all'}
             label={this.getGrommetIntlMessage('All')}
             checked={this.state.data[attribute.attribute].all}
             onChange={this._onChangeAll
               .bind(this, attribute.attribute, attribute.filter)} />
         );
-        return components.concat(values);
+        return (<fieldset>
```
Comment (alansouzati) [inline src/js/components/index/IndexFilters.js]: can you add `<fieldset className="index-filters">` and create a scss file for this?

**Comment by alansouzati** [comment]
Thanks Tiago, I will update documentation properly for `ariaDescribedby`

**Pull request merged**